### PR TITLE
Backport #4615 (C++ header ambiguity fix) to 1.4

### PR DESCRIPTION
**Pull request by Manishearth**
Backports https://github.com/unicode-org/icu4x/pull/4615 to 1.4

### CODE REVIEW

**Comment by Manishearth** [comment]
Would like some backport approval:

 - [ ] @robertbastian 
 - [ ] @hsivonen 
 - [ ] @zbraniecki
 - [ ] @makotokato 
 - [x] @sffc

File: ffi/capi/dart/package/lib/src/lib.g.dart [hunk 1]
```diff
@@ -9720,12 +9720,10 @@ class ScriptWithExtensionsBorrowed implements ffi.Finalizable {
   // ignore: non_constant_identifier_names
   static final _ICU4XScriptWithExtensionsBorrowed_get_script_extensions_val =
       _capi<
-                  ffi
```
Comment (sffc) [inline ffi/capi/dart/package/lib/src/lib.g.dart:9723]: Why is there a diff in a .dart file?

Comment (Manishearth) [reply ffi/capi/dart/package/lib/src/lib.g.dart:9723]: dart regen runs dart fmt

**Comment by Manishearth** [comment]
It's concerning that CI doesn't run, build-test has this config and it ought to work:

```yaml
on:
  push:
    branches: [ main ]
  pull_request:
    branches: '*'
  workflow_dispatch:
    inputs:
      nightly:
        description: "Test on latest nightly"
        required: true
        default: false
        type: boolean
  schedule:
    # Run nightly cronjob CI every day at 14 UTC / 6AM PST / 3PM CET
    - cron:  '0 14 * * *'
```

**Comment by Manishearth** [comment]
Fixed it.

File: ffi/capi/dart/package/lib/src/lib.g.dart (hunk 1 above)
Comment (Manishearth) [reply ffi/capi/dart/package/lib/src/lib.g.dart]: Looks like CI pins a different dart compiler. Undone.

**Comment by Manishearth** [comment]
I'm ok with test-tutorials being broken it's a weird job anyway

**Comment by Manishearth** [comment]
Wrong label. But we should definitely discuss it in today's call.

**Review by sffc — APPROVED** [review]

**Review by younies — APPROVED** [review]

**Pull request merged**